src/layout/ErrorModal.tsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2021 PixieBrix, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { faExclamationCircle } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import React, { useState } from "react";
+import { Button, Modal, ModalBody } from "react-bootstrap";
+import { useHistory } from "react-router";
+
+const errorMessages = new Map([
+  [
+    "ERR_BROWSER_ACTION_TOGGLE_SPECIAL_PAGE",
+    "PixieBrix can’t run on internal browser pages",
+  ],
+  ["ERR_BROWSER_ACTION_TOGGLE", "PixieBrix could not run on the page"],
+]);
+
+const ErrorModal: React.FunctionComponent = () => {
+  const [show, setShow] = useState(true);
+  const history = useHistory();
+  const message = errorMessages.get(
+    new URLSearchParams(location.search).get("error")
+  );
+
+  const handleClose = () => {
+    setShow(false);
+    const params = new URLSearchParams(location.search);
+    params.delete("error");
+    history.replace({ search: params.toString() });
+  };
+
+  if (message) {
+    return (
+      <Modal show={show} onHide={handleClose}>
+        <Modal.Header closeButton>
```

Comment:
I think the close button is redundant here

<img width="538" alt="Screen Shot" src="https://user-images.githubusercontent.com/1402241/127271903-90cf24ff-47db-41df-935c-0f084bc63f8c.png">

```suggestion
        <Modal.Header>
```

<img width="554" alt="Screen Shot 15" src="https://user-images.githubusercontent.com/1402241/127271919-477f4ba6-9099-44cf-b5e6-07e5850d48f2.png">

Reply:
> I think the close button is redundant here

Cons: having two ways of doing something can add cognitive burden on the user. Pros: it's clearly not the primary button action, so it's not hurting much

I would probably leave the upper "x" but center the "Close" button?

/bikeshedding

Reply:
I think right-aligned buttons are common from what I see https://uxplanet.org/how-to-write-good-error-messages-858e4551cd4 (screenshots only)

Reply:
> I would probably leave the upper "x" but center the "Close" button?

I originally added the close button because the modal looked a little bare without something in the footer. So I'm sort of erring on the side of keeping the close button and removing the "x".

Reply:
<img width="521" alt="Screen Shot 2021-07-28 at 8 22 07 AM" src="https://user-images.githubusercontent.com/36575242/127349497-3bf79964-939e-4669-bf70-3bc093076e97.png">
Actually, now that I'm looking at it again it looks fine. Without the button it looks less "urgent," which might be ideal for something like this.

Reply:
I'm ok with this as long as clicking outside the modal also exits the modal (there should be a prop on React bootstrap's modal that controls that)

Reply:
One problem with the "Close" button is it's ambiguous as to whether it closes the extension page or just the modal. So I like the "x"

Reply:
As much as I'd like one less button, I think button-less popups are not the way to go.  The original one was good 😅

Reply:
> The original one was good 😅

Thanks for the feedback 😂 We decided to go with the original, but with button "primary" instead of "danger". Important things to discuss, anyways!